app.go

```diff
@@ -706,16 +798,26 @@ func (app *App) run(done <-chan os.Signal) {
 	defer cancel()
 
 	if err := app.Start(startCtx); err != nil {
-		app.logger.Fatalf("ERROR\t\tFailed to start: %v", err)
+		fxlog.Error("failed to start", fxlog.Field{
+			Key:   "error",
+			Value: err,
+		}).Write(app.log)
+		// TODO: add option to override os.Exit behavior.
+		os.Exit(1)
 	}
 
-	app.logger.PrintSignal(<-done)
+	fxlog.Info(strings.ToUpper((<-done).String())).Write(app.log)
```

Comment:
I would change this to "received signal" and then make the signal a field of the log statement.

And just for readability, read from channel should be separate from stringifying it and logging it.